scripts/check-codegen.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+
+# This script is meant for CI to ensure that codegen files are up-to-date.
+#
+# Because the codegen script is pretty slow, it bails early if no files in pkg/apis
+# were changed by the commit CI is running against (as compared to HEAD).
+#
+# If there were changed files, the codegen script is run and if there are any uncommitted
+# files as a result, it fails with an error with a hint to run the script locally and push.
+
+set -e
+
+DIR=$(dirname "$0")
+
+CODEGEN_REGEX="^pkg/apis|^pkg/openapi"
+
+function print_file_list() {
+    while IFS= read -r line
+    do
+      if [[ $line =~ $CODEGEN_REGEX ]]; then
+        printf "  - %s\n" "${line}"
+      fi
+    done < <(printf '%s\n' "${*}")
+}
+
+changes=""
+if [[ -n "${CIRCLECI}" ]]; then
```

Comment:
hmmmm...would it be simpler to simply verify if any files are modified at the beginning?

not totally sure we need circleci-specific logic here